diff --git a/node_modules/hap-nodejs/index.d.ts b/node_modules/hap-nodejs/index.d.ts
index ee44b7e..7e26bb6 100644
--- a/node_modules/hap-nodejs/index.d.ts
+++ b/node_modules/hap-nodejs/index.d.ts
@@ -31,6 +31,10 @@ declare namespace HAPNodeJS {
         characteristics: Characteristic[];
         optionalCharacteristics: Characteristic[];
 
+        isHiddenService: boolean;
+        isPrimaryService: boolean;
+        linkedServices: Service[];
+
         addCharacteristic(characteristic: Characteristic | Function): Characteristic;
         setHiddenService(isHidden: boolean): void;
         addLinkedService(newLinkedService: Service | Function): void;
@@ -158,6 +162,14 @@ declare namespace HAPNodeJS {
     export interface Characteristic extends IEventEmitterCharacteristic {
         new (displayName: string, UUID: string, props?: CharacteristicProps): Characteristic;
 
+        displayName: string;
+        UUID: string;
+        iid: string;
+        value: CharacteristicValue;
+        eventOnlyCharacteristic: boolean
+        props: CharacteristicProps;
+
+
         Formats: typeof Characteristic.Formats;
         Units: typeof Characteristic.Units;
         Perms: typeof Characteristic.Perms;
diff --git a/node_modules/hap-nodejs/lib/Service.js b/node_modules/hap-nodejs/lib/Service.js
index 749e93d..e9d2f47 100644
--- a/node_modules/hap-nodejs/lib/Service.js
+++ b/node_modules/hap-nodejs/lib/Service.js
@@ -52,6 +52,10 @@ function Service(displayName, UUID, subtype) {
   // if one was given
   // if you don't provide a display name, some HomeKit apps may choose to hide the device.
   if (displayName) {
+    // Don't add the `Name` characteristic to `Service.AccessoryInformation` since it will be added when defining that service.
+    if (UUID === '0000003E-0000-1000-8000-0026BB765291') {
+      return;
+    }
     // create the characteristic if necessary
     var nameCharacteristic =
       this.getCharacteristic(Characteristic.Name) ||
